haskell/private/haskell_impl.bzl

```diff
@@ -179,14 +189,46 @@ def _haskell_binary_common_impl(ctx, is_test):
         bin_info = bin_info,
     )
 
+    executable = binary
+
+    if inspect_coverage:
+        binary_path = paths.join(ctx.workspace_name, binary.short_path)
+        hpc_path = paths.join(ctx.workspace_name, hs.toolchain.tools.hpc.short_path)
+        tix_file_path = hs.label.name + ".tix"
+        mix_file_paths = ""
+        for m in conditioned_mix_files:
+            mix_file_paths += paths.join(ctx.workspace_name, m.short_path) + " "
+        expected_expression_coverage = ctx.attr.expected_expression_coverage
+        wrapper = hs.actions.declare_file("coverage_wrapper.sh")
+        ctx.actions.expand_template(
+            template = ctx.file._coverage_wrapper_template,
+            output = wrapper,
+            substitutions = {
+                "{binary_path}": binary_path,
+                "{hpc_path}": hpc_path,
+                "{tix_file_path}": tix_file_path,
+                "{expected_expression_coverage}": str(expected_expression_coverage),
+                "{mix_file_paths}": mix_file_paths,
+            },
+            is_executable = True,
+        )
+        executable = wrapper
+
     return [
         build_info,
         bin_info,
         DefaultInfo(
-            executable = binary,
+            executable = executable,
             files = target_files,
             runfiles = ctx.runfiles(
-                files = solibs,
+                files =
+                    solibs +
+                    c.conditioned_mix_files +
+                    [
+                        ctx.file._bash_runfiles,
+                        hs.toolchain.tools.hpc,
+                        binary,
+                    ],
```

Comment:
These are unconditionally added, right? Can we only add them when `inspect_coverage` is `True`?